lib/config/presets/internal/helpers.ts

```diff
@@ -30,4 +30,15 @@ export const presets: Record<string, Preset> = {
       },
     ],
   },
+  pinGitHubActionDigestsToSemver: {
+    description: 'Convert pinned GitHub Action digests to SemVer.',
+    packageRules: [
+      {
+        extends: ['helpers:pinGitHubActionDigests'],
+        extractVersion: '^(?<version>v\\d+\\.\\d+\\.\\d+)$',
+        versioning:
+          'regex:^v(?<major>\\d+)(\\.(?<minor>\\d+)\\.(?<patch>\\d+))?$',
```

Comment:
Do you want to support v1.2.3, v1.2, v1? If yes:
```suggestion
          'regex:^v(?<major>\\d+)(\\.(?<minor>\\d+))?(\\.(?<patch>\\d+))?$',
```
If not:
```suggestion
          'regex:^v(?<major>\\d+)\\.(?<minor>\\d+)\\.(?<patch>\\d+)$',
```

Reply:
and the same should probably be done to `extractVersion`?

Reply:
Huh, based on https://semver.org/ all 3 components are required, but IRL not all GitHub actions may follow the 3-component versioning.

Reply:
> but IRL not all GitHub actions may follow the 3-component versioning.

Many actions that I've seen have two "version options", `v3` or `v3.0.0`.

- `v3`: give me the latest action in the `v3` range
- `v3.0.0`: give me this specific action, and let me update to a new version manually or via Renovate

Reply:
Yeah, that's normal and *recommended*, but enforced in any way.

Examples:
https://github.com/szenius/set-timezone/tags 
https://github.com/npalm/action-docs-action/tags

These repos have open issues to use the vX tags, but not yet. And the first one uses 2 components only.

Reply:
Looks like the old "garbage in, garbage out" problem, where we have to work-around projects using silly versions.

I'll let the maintainers decide what to do here.

Reply:
Considering the name "Pin...ToSemver", requiring 3 makes sense, but following the principle of least astonishment, `3.0` should "just work" the same way as `3.0.0`.

Reply:
@HonkingGoose you might want to leave this open for visibility of maintainer reviews.

Reply:
Let's go with 3 components

Reply:
@TWiStErRob do you want to suggest the correct regexes for 3 component versions? 🙃

Reply:
😂, just remove the question-mark and then the redundant parentheses, and unit test!

Reply:
Oh I already applied your suggestion:

- https://github.com/renovatebot/renovate/pull/23663/commits/938a4f5515751705ef3664369a85ea81bd837fa3

Reply:
Yeah reverse that one, because Rhys wants only 1 and 3 components only.